### DISCUSSION TITLE
Libero Place and Route Error

### DISCUSSION THREAD

**Discussion by Shivaraj199** [polarfire-soc/polarfire-soc-documentation#278] [General] [answered]
Hello ,
I am trying to do Place and route in Libero but Implement design and syntesize tool always fails. I thought there might be an Issue with my design so I tried with the reference design from this link https://github.com/polarfire-soc/icicle-kit-reference-design.git and I am still getting the same FlexNet Licensing error:-25,234
I created a symbolic link  '/usr/tmp/tmp' from the command sudo ln -s /tmp /usr/tmp . I am sure libero finds the license file as I can open the Libero but uncertain why synthesize tool fails.
I am using ubuntu 20.04 
 

Libero Version: 2022.2.0.10
Release: v2022.2

The Log file is 
Synplify Pro (R) 
Starting:    /home/xyz/microchip/Libero_SoC_v2022.2/SynplifyPro/linux_a_64/mbin/synbatch
Install:     /home/xyz/microchip/Libero_SoC_v2022.2/SynplifyPro
Hostname:    io-rtx-portable
Date:        Thu Jan  5 15:01:14 2023
Version:     S-2021.09M-SP1

Arguments:   -product synplify_pro -licensetype synplifypro_actel -batch -log synplify.log MPFS_ICICLE_KIT_BASE_DESIGN_syn.tcl
ProductType: synplify_base

License server system does not support this version of this feature.
Feature:       synplifypro_actel
License path:  /home/xyz/flexlm/License.dat:
FlexNet Licensing error:-25,234

Host ID:  "xxxxxxxxxxxx yyyyyyyyyyyy"
Please contact support at https://solvnetplus.synopsys.com/ManageCase?ccf=1

Running in Vendor Mode

**Comment by TommyMurphyTM1234** [comment]
There's something wrong with your license setup.
In my experience, setting up the licensing for Libero SoC (and associated tools) on Linux is not straightforward.
And the Microchip documentation isn't that clear, helpful or consistent in many places.
This real-world blog describing the setup might be worth reading through.

* https://www.cnx-software.com/2021/10/25/installing-libero-soc-in-ubuntu-and-windows-10/

**Reply by Shivaraj199** [reply]
Thanks for the reply I will look into  that

**Comment by andrenemat** [comment, marked as answer]
Are you using a node locked license (locked to diskID) or floating license (locked to server MAC address). In Linux you can only use a floating license. Please review the document at the link posted by [TommyMurphyTM1234]. Also please double check that the latest versions of the Daemons are installed on your server.
https://www.microchip.com/en-us/products/fpgas-and-plds/fpga-and-soc-design-tools/fpga/licensing

**Reply by Shivaraj199** [reply]
I am using floating license, As I can see that Libero has can detect the License once I start up the server

**Reply by TommyMurphyTM1234** [reply]
> I am using floating license, As I can see that Libero has can detect the License once I start up the server

If Libero licensing is working OK but Synplify is not then perhaps you installed the 64-bit license daemons?

According to this page

* https://www.microchip.com/en-us/products/fpgas-and-plds/fpga-and-soc-design-tools/fpga/licensing

Under "Daemons Downloads"...

> These 64-bit licensing Daemons support Microchip's Libero Standalone tools only. To work with Synopsys and Mentor tools, install 32-bit licensing Daemons.

**Comment by eppidei** [comment]
you may find it useful to have a look at this guide
https://www.microchip.com/en-us/products/fpgas-and-plds/fpga-and-soc-design-tools/fpga/libero-software-later-versions#Documentation
Then go to Licensing and Installation User Guides -> [How to Set Up Your Linux Environment for Libero SoC Design Suite Versions 2022.2](https://www.microchip.com/content/dam/mchp/documents/FPGA/core-docs/Libero/libero/linux_setup_libero_soc_ug.pdf)
